stubs/bleach/bleach/html5lib_shim.pyi

```diff
@@ -1,30 +1,59 @@
-from _typeshed import Incomplete
+import re
 from collections.abc import Generator, Iterable, Iterator
+from typing import Any
 
-class HTMLParser:  # actually html5lib.HTMLParser
-    def __getattr__(self, __name: str) -> Incomplete: ...
+# We don't re-export any `html5lib` types / values here, because they are not
+# really public and may change at any time. This is just a helper module,
+# import things directly from `html5lib` instead!
+from html5lib import HTMLParser
+from html5lib._inputstream import HTMLUnicodeInputStream
+from html5lib._tokenizer import HTMLTokenizer
+from html5lib.serializer import HTMLSerializer
 
-class Filter:  # actually html5lib.filters.base.Filter
-    source: Incomplete
-    def __init__(self, source) -> None: ...
-    def __iter__(self) -> Iterator[Incomplete]: ...
-    def __getattr__(self, name: str) -> Incomplete: ...  # copy attributes from source
+HTML_TAGS: list[str]
+HTML_TAGS_BLOCK_LEVEL: frozenset[str]
+AMP_SPLIT_RE: re.Pattern[str]
+ENTITIES: dict[str, str]
+TAG_TOKEN_TYPES: set[int]
+TAG_TOKEN_TYPE_CHARACTERS: int
+TAG_TOKEN_TYPE_END: int
+TAG_TOKEN_TYPE_PARSEERROR: int
+TAG_TOKEN_TYPE_START: int
```

Comment:
I know this PR predates the resolution of #7258 , but I'm pretty certain these are all meant to be constant. They could be marked as `Final`.
`HTML_TAGS` is also a `frozenset`
```suggestion
from typing_extensions import Final

HTML_TAGS: Final[frozenset[str]]
HTML_TAGS_BLOCK_LEVEL: Final[frozenset[str]]
AMP_SPLIT_RE: Final[re.Pattern[str]]
ENTITIES: Final[dict[str, str]]
TAG_TOKEN_TYPES: Final[set[int]]
TAG_TOKEN_TYPE_CHARACTERS: Final[int]
TAG_TOKEN_TYPE_END: Final[int]
TAG_TOKEN_TYPE_PARSEERROR: Final[int]
TAG_TOKEN_TYPE_START: Final[int]
```